### PR TITLE
Backport pacman updates to 1.9 [take 2]

### PR DESCRIPTION
**Pull request by indrajitr**
~~Backport pacman PRs #1037 to stable-1.9.
Please apply this __after__ #1028.~~
Backport selective pacman PRs #480, #762, #807, #1018 to stable-1.9.
This __replaces__ #1028.

NB: Description updated based on feedback in (https://github.com/ansible/ansible-modules-extras/pull/1038#issuecomment-143894698).

### CODE REVIEW

**Comment by bcoca** [comment]
this includes several 2.0 features, we do not backport features, only bugfixes

**Comment by indrajitr** [comment]
Sure. Cherry-picked the specific ones that either fell through the crack when `stable-1.9` was created and/or the ones that actually fix bugs (or incorrect behavior).

**Comment by robynbergeron** [comment]
Hi @indrajitr -- thanks for the cherry-picking, and the PR in general :)

Marking for inclusion as owner_pr; cc @ansible/core team -- @bcoca, you looked at this one previously, so hopefully this works out.

File: packaging/os/pacman.py [hunk 1]
```diff
@@ -59,7 +63,14 @@
             - Whether or not to refresh the master package lists. This can be
               run as part of a package installation or as a separate step.
         required: false
-        default: "no"
+        default: no
+        choices: ["yes", "no"]
+
+    upgrade
```
Comment (bcoca) [inline packaging/os/pacman.py]: this is still a 2.0 feature

**Comment by indrajitr** [comment]
@bcoca, right you are. It was an incorrect rebase. Removed `upgrade` with a separate commit.

**Comment by gregdek** [comment]
Putting back into core_review. Which is, I believe, the proper endpoint for all backport requests. cc: @bcoca for re-review.

**Comment by robynbergeron** [comment]
And I'm just adding the backport label because it is so. Still pending @ansible/core review :) Thanks for your patience, @indrajitr! :)

**Pull request closed without merging**